x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest, SavedObjectsClientContract } from 'kibana/server';
+import { i18n } from '@kbn/i18n';
+import { Logger } from 'kibana/server';
+import { MlClient } from '../ml_client';
+import {
+  AnomalyDetectionJobsHealthRuleParams,
+  JobSelection,
+} from '../../routes/schemas/alerting_schema';
+import { datafeedsProvider, DatafeedsService } from '../../models/job_service/datafeeds';
+import { ALL_JOBS_SELECTION } from '../../../common/constants/alerts';
+import { MlJobsResponse } from '../../../common/types/job_service';
+import { DatafeedStats } from '../../../common/types/anomaly_detection_jobs';
+import { GetGuards } from '../../shared_services/shared_services';
+import { AnomalyDetectionJobsHealthAlertContext } from './register_jobs_monitoring_rule_type';
+import { getResultJobsHealthRuleConfig } from '../../../common/util/alerts';
+
+interface TestResult {
+  name: string;
+  context: AnomalyDetectionJobsHealthAlertContext;
+}
+
+type TestsResults = TestResult[];
+
+type NotStartedDatafeedResponse = Array<DatafeedStats & { job_id: string }>;
+
+export function jobsHealthServiceProvider(
+  mlClient: MlClient,
+  datafeedsService: DatafeedsService,
+  logger: Logger
+) {
+  /**
+   * Extract result list of job ids based on included and excluded selection of jobs and groups.
+   * @param includeJobs
+   * @param excludeJobs
+   */
+  const getResultJobIds = async (includeJobs: JobSelection, excludeJobs?: JobSelection | null) => {
+    const jobAndGroupIds = [...(includeJobs.jobIds ?? []), ...(includeJobs.groupIds ?? [])];
+
+    const includeAllJobs = jobAndGroupIds.some((id) => id === ALL_JOBS_SELECTION);
+
+    // Extract jobs from group ids and make sure provided jobs assigned to a current space
+    const jobsResponse = (
+      await mlClient.getJobs<MlJobsResponse>({
+        ...(includeAllJobs ? {} : { job_id: jobAndGroupIds.join(',') }),
```

Comment:
you might be able to pass in the `jobAndGroupIds` without the join here. i believe the new es client allows it

Reply:
removed in https://github.com/elastic/kibana/pull/106084/commits/2a1eb8245b17d6c7ea93fe7a7896e4c93884e7ba